### PR TITLE
HPCC-22927 RemotePull that needs external Esp should optionally take credentials

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add username/password to Std.File.RemotePull() function

Implement code to handle username/password

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested manually between two systems where on the target the LDAP authentication was enabled.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review.

File: plugins/fileservices/fileservices.cpp [hunk 1]
```diff
@@ -2248,6 +2250,15 @@ FILESERVICES_API char * FILESERVICES_CALL fsfRemotePull_impl(ICodeContext *ctx,
 
     req->setExpireDays(expireDays);
 
+    // Store username/psw
```
Comment (jakesmith) [inline plugins/fileservices/fileservices.cpp:2253]: perhaps a bit misleading to say 'Store' here, it mighy imply it is permanent and a security issue, whereas really it's just setting into an object used by the request.

File: plugins/fileservices/fileservices.cpp [hunk 2]
```diff
@@ -2248,6 +2250,15 @@ FILESERVICES_API char * FILESERVICES_CALL fsfRemotePull_impl(ICodeContext *ctx,
...
+    server.setUsernameToken(username, userPw, nullptr);
```
Comment (jakesmith) [inline plugins/fileservices/fileservices.cpp:2254]: should this be after the if below?

**Comment by jakesmith** [comment]
@AttilaVamos - looks good, just a couple of v. minor comments.

File: plugins/fileservices/fileservices.cpp (hunk 1 above)
Comment (AttilaVamos) [reply plugins/fileservices/fileservices.cpp]: Changed

File: plugins/fileservices/fileservices.cpp (hunk 2 above)
Comment (AttilaVamos) [reply plugins/fileservices/fileservices.cpp]: Moved

**Comment by AttilaVamos** [comment]
@jakesmith updated, please take a quick look.

**Comment by jakesmith** [comment]
@AttilaVamos looks good.

@richardkchapman - ready to merge.

**Pull request merged**